cassiopeia/simulator/Cas9LineageTracingDataSimulator.py

```diff
@@ -132,42 +134,66 @@ def __init__(
         self.size_of_cassette = size_of_cassette
         self.number_of_cassettes = number_of_cassettes
         self.collapse_sites_on_cassette = collapse_sites_on_cassette
+        number_of_characters = size_of_cassette * number_of_cassettes
 
         if isinstance(mutation_rate, float):
             if mutation_rate < 0:
                 raise DataSimulatorError(
                     "Mutation rate needs to be" " non-negative."
                 )
-            number_of_characters = size_of_cassette * number_of_cassettes
             self.mutation_rate_per_character = [
                 mutation_rate
             ] * number_of_characters
         else:
-            if len(mutation_rate) != (
-                self.number_of_cassettes * self.size_of_cassette
-            ):
+            if len(mutation_rate) == number_of_characters:
+                self.mutation_rate_per_character = mutation_rate
+            elif len(mutation_rate) == self.size_of_cassette:
+                self.mutation_rate_per_character = (
+                    mutation_rate * self.number_of_cassettes
```

Comment:
you should mke sure that `mutation_rate` is actually a List. If it's a numpy array, then this line will just multiply the mutation rate by the number of cassettes.

Reply:
Good catch!